src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -591,140 +572,134 @@ private void WriteGreenRewriter()
+
+                    Unindent();
                 }
             }
-            WriteLine("  }");
+
+            CloseBlock();
         }
 
         private void WriteContextualGreenFactories()
         {
             var nodes = Tree.Types.Where(n => !(n is PredefinedNode) && !(n is AbstractNode)).ToList();
             WriteLine();
-            WriteLine("  internal partial class ContextAwareSyntax");
-            WriteLine("  {");
-
-            WriteLine();
-            WriteLine("    private SyntaxFactoryContext context;");
+            WriteLine("internal partial class ContextAwareSyntax");
+            OpenBlock();
             WriteLine();
+            WriteLine("private SyntaxFactoryContext context;");
 
             WriteLine();
-            WriteLine("    public ContextAwareSyntax(SyntaxFactoryContext context)");
-            WriteLine("    {");
-            WriteLine("        this.context = context;");
-            WriteLine("    }");
+            WriteLine("public ContextAwareSyntax(SyntaxFactoryContext context)");
+            WriteLine("    => this.context = context;");
 
             WriteGreenFactories(nodes, withSyntaxFactoryContext: true);
-
-            WriteLine("  }");
+            CloseBlock();
         }
 
         private void WriteStaticGreenFactories()
         {
             var nodes = Tree.Types.Where(n => !(n is PredefinedNode) && !(n is AbstractNode)).ToList();
             WriteLine();
-            WriteLine("  internal static partial class SyntaxFactory");
-            WriteLine("  {");
-
+            WriteLine("internal static partial class SyntaxFactory");
+            OpenBlock();
             WriteGreenFactories(nodes);
-
             WriteGreenTypeList();
-
-            WriteLine("  }");
+            CloseBlock();
         }
 
         private void WriteGreenFactories(List<TreeType> nodes, bool withSyntaxFactoryContext = false)
         {
             for (int i = 0, n = nodes.Count; i < n; i++)
             {
                 var node = nodes[i];
+                WriteLine();
                 this.WriteGreenFactory((Node)node, withSyntaxFactoryContext);
-                if (i < n - 1)
-                    WriteLine();
             }
         }
 
         private void WriteGreenTypeList()
         {
             WriteLine();
-            WriteLine("    internal static IEnumerable<Type> GetNodeTypes()");
-            WriteLine("    {");
-            WriteLine("        return new Type[] {");
+            WriteLine("internal static IEnumerable<Type> GetNodeTypes()");
+            Indent();
+            WriteLine("=> new Type[]");
+            OpenBlock();
 
             var nodes = Tree.Types.Where(n => !(n is PredefinedNode) && !(n is AbstractNode)).ToList();
             for (int i = 0, n = nodes.Count; i < n; i++)
             {
                 var node = nodes[i];
-                Write("           typeof({0})", node.Name);
-                if (i < n - 1)
-                    Write(",");
```

Comment:
constructs like array initializers can have commas after every item.  So this just simplified things by always emitting the comma.  We do the same thing for switch expressions when we use them.